lease/lessor.go

```diff
@@ -412,7 +419,8 @@ func (le *lessor) Stop() {
 func (le *lessor) runLoop() {
 	defer close(le.doneC)
 
-	for {
+	revokerLimiter := rate.NewLimiter(rate.Limit(leaseRevokesPerSecond), leaseRevokesPerSecond)
+	for revokerLimiter.Wait(context.Background()) == nil {
```

Comment:
I don't think this will work since `findExpiredLeases` can give `len(ls) > leaseRevokesPerSecond`. Maybe just have something like `if len(ls) > leaseRevokesPerSecond/2 { ls = ls[:leaseRevokesPerSecond/2]}` before posting to `expiredC`?